### DISCUSSION TITLE
Upgrade

### DISCUSSION THREAD

**Discussion by mattyg2301** [postalserver/postal#2901] [Help with using Postal]
Good day all,

I noticed version 3 is out and after trying to follow the upgrade command as per the documentation I get the below error:

/opt/postal/install# git pull origin
warning: redirecting to https://github.com/postalserver/install/
Auto-merging templates/docker-compose.yml
CONFLICT (modify/delete): templates/docker-compose.v3.yml deleted in HEAD and modified in c82658da7fa16615a2fba9bae2d02c860816fdb3. Version c82658da7fa16615a2fba9bae2d02c860816fdb3 of templates/docker-compose.v3.yml left in tree.
CONFLICT (modify/delete): examples/postal.v3.yml deleted in HEAD and modified in c82658da7fa16615a2fba9bae2d02c860816fdb3. Version c82658da7fa16615a2fba9bae2d02c860816fdb3 of examples/postal.v3.yml left in tree.
Auto-merging bin/postal
CONFLICT (content): Merge conflict in bin/postal
Automatic merge failed; fix conflicts and then commit the result.

Not sure how to resolve, any advise would be greatly appreciated.

**Comment by willpower232** [comment]
It sounds like you made some changes to your local installer which git is struggling with.

I would `git merge --abort` to put you back as you were and then try and identify the changes, perhaps with `git status` or `git diff`.

**Reply by mattyg2301** [reply]
Thanks for the speedy response, I'm not really sure what that means to be honest, my GIT knowledge is very limited.

As far as I'm aware, I haven't made any changes to anything locally that I can recall.

The last thing I did was run an upgrade about a month ago that went without issues and when I ran postal upgrade today I received the below:

postal upgrade
warning: redirecting to https://github.com/postalserver/install/
remote: Enumerating objects: 80, done.
remote: Counting objects: 100% (80/80), done.
remote: Compressing objects: 100% (34/34), done.
remote: Total 71 (delta 36), reused 56 (delta 26), pack-reused 0
Unpacking objects: 100% (71/71), 20.04 KiB | 526.00 KiB/s, done.
From https://postalserver.io/start/install
 * branch            main       -> FETCH_HEAD
 + 74a95e4...c82658d main       -> origin/main  (forced update)
Auto-merging templates/docker-compose.yml
CONFLICT (modify/delete): templates/docker-compose.v3.yml deleted in HEAD and modified in c82658da7fa16615a2fba9bae2d02c860816fdb3. Version c82658da7fa16615a2fba9bae2d02c860816fdb3 of templates/docker-compose.v3.yml left in tree.
CONFLICT (modify/delete): examples/postal.v3.yml deleted in HEAD and modified in c82658da7fa16615a2fba9bae2d02c860816fdb3. Version c82658da7fa16615a2fba9bae2d02c860816fdb3 of examples/postal.v3.yml left in tree.
Auto-merging bin/postal
CONFLICT (content): Merge conflict in bin/postal
Automatic merge failed; fix conflicts and then commit the result.

**Reply by mattyg2301** [reply]
Just adding the output of git status

git status
On branch main
Your branch and 'origin/main' have diverged,
and have 3 and 17 different commits each, respectively.
  (use "git pull" to merge the remote branch into yours)

Untracked files:
  (use "git add <file>..." to include in what will be committed)
        FETCH_HEAD

nothing added to commit but untracked files present (use "git add" to track)

**Reply by mattyg2301** [reply]
Hi Will,

Looks like I'm all up to date now, all I did was run apt update && apt upgrade, rebooted and when I ran postal stop and then postal start it pulled the latest release down and is up and running.

Thanks for you help earlier!
